### ISSUE TITLE
Map contribution: “3 × 3 = 9”

### ISSUE THREAD

**Issue by nmoskopp** [OpenHV/OpenHV#642]
![hv-2022-11-06T193812392Z](https://user-images.githubusercontent.com/22373642/200191353-af712a45-750d-45cc-8ad6-70c6fc308d64.png)

This small map features several islands on a grid 3 high and 3 wide. The middle island is partitioned so that units can shoot at each other, but not cross over to the other side. The map also features pre-built neutral walls around resources and money-generating buildings. Creep mines are placed on damaged tiles in the partitioned middle island. Icebergs and deep ocean are included to prevent players from blockading the middle island using ships.

The map was designed for short sessions with lots of tension. Because of this, resources and money-generating buildings lie directly on the land path to the enemy or are reachable by artillery from an island nearby. There are few resources on the map overall, to force players to take the initiative.

As GitHub does not take arbitrary extensions, you must rename `339.oramap.zip` to `339.oramap` to use the map.

[339.oramap.zip](https://github.com/OpenHV/OpenHV/files/9946679/339.oramap.zip)

**Comment by Mailaender** [comment]
Can you send a pull request?

**Comment by Mailaender** [comment]
You can also share maps at https://resource.openra.net/maps/

**Comment by Mailaender** [comment]
Not suitable due to https://github.com/OpenHV/OpenHV/issues/640. All maps have to be playable by the AI.

**Comment by nmoskopp** [comment]
> Not suitable due to #640. All maps have to be playable by the AI.

This map is not affected by #640, as buildings can not be placed on rails.

Please reopen and reconsider or show me where #640 can occur on this map.

**Comment by SeanScherer** [comment]
Looks interesting. I'll give it a spin :). If you like, I'll upload it to OpenRA.net for you (assuming the AI doesn't block itself, as you say). 
Matthias, I think you may have missed that rails were added for the bridges here (to make them unbuildable, ie to address #640 ).

**Comment by SeanScherer** [comment]
Definitely an unusual take on OpenHV :). 

For the sake of human players, you might consider rails towards the middle island (though perhaps you consider the option of "blocking" passage - ie for the enemy - intentional).

As said, happy to upload if you want me to!

**Comment by nmoskopp** [comment]
> Definitely an unusual take on OpenHV :).

I literally made it because the map “Business as Usual” became too boring. :wink:

**Comment by Mailaender** [comment]
If you don't know how to use git and pull requests, join the chat.

**Comment by nmoskopp** [comment]
> > Testing map: 3 × 3 = 9
> > OpenRA.Utility(1,1): Error: This map does not define a valid cordon.
> > A one cell (or greater) border is required on all four sides between the playable bounds and the map edges

What did you do to get this message and what could I do to fix it?

**Comment by Mailaender** [comment]
I ran `make test` like the CI checks would if you had submitted a pull request. You need to change the map boundaries. It seems you manually changed them. One shrouded cell line is required for units to appear.

**Comment by nmoskopp** [comment]
> It seems you manually changed them.

I did indeed do that. By accident, I made the map 1 tile less wide and high than I wanted it to be, so I edited the YAML file manually. It still played just fine.

How could I enlarge a map properly?

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/wiki/Utility has a map resize option. But I think it has the same problems as #645.

**Comment by nmoskopp** [comment]
I do not think it has the same problems, as I have found no path blocking issues on it after I added the rails.

**Comment by nmoskopp** [comment]
@Mailaender I have found no pathfinding or AI-blocking-itself issues. I resized the map. Does that work for you?

```
; ./OpenHV-20221030-x86_64.AppImage --utility --resize-map /home/nmoskopp/.config/openra/maps/hv/20221030/339.oramap 58 58
Resizing map 3 × 3 = 9 from 57,57 to 58,58
# exited 0
```
[339.oramap.zip](https://github.com/OpenHV/OpenHV/files/9985261/339.oramap.zip)

**Comment by SeanScherer** [comment]
Hey, I've not seen what licence you'd be releasing under.

I'd like to upload to https://resource.openra.net/ - which would make it easier to setup a multiplayer game. (which might not end up being a step closer to official inclusion, but I feel it at least wouldn't hurt).

I need to specify a licence though if I'm doing that though. ( fyi, OpenHV's default map licence seems to be: Creative Commons Attribution ShareAlike 4.0).

(and sorry for duplication, I know I mentioned this in my email as well).

**Comment by nmoskopp** [comment]
@SeanScherer the license for  “3 × 3 = 9” is, at your option, Creative Commons Attribution ShareAlike 4.0+ and/or GPL 3+.

**Comment by SeanScherer** [comment]
Cool, and thank you :). Saw that yesterday, but just uploaded it now : https://resource.openra.net/maps/48875 !

(Unfortunately, I can't get the official server to switch to it atm - I had the same issue with a different map I uploaded recently...  :/ . Will be digging into it! )

**Comment by SeanScherer** [comment]
I've filed an issue: #651
